main.py

```diff
@@ -40,156 +38,130 @@
+
+  return brand_assessment
+
+
+def main(arg_list: list[str] | None = None) -> None:
+  """Main ABCD Assessment execution. See docstring and args.
 
-    video_uris = expand_uris(config.video_uris)
-
-    for video_uri in video_uris:
-        print(f"\n\nProcessing ABCD Assessment for video {video_uri}... \n")
-
-        # 1) Prepare video
-        trim_video(config, video_uri)
-
-        # Check size of video to avoid processing videos > 7MB
-        video_metadata = get_blob(video_uri)
-        size_mb = video_metadata.size / 1e6
-        if config.use_llms and size_mb > config.video_size_limit_mb:
-            print(
-                f"The size of video {video_uri} is greater than {config.video_size_limit_mb} MB. Skipping execution."
-            )
-            continue
-
-        # 3) Execute ABCD Assessment
-        video_assessment = {
-            "video_uri": video_uri,
-        }
-
-        if config.use_annotations:
-            generate_video_annotations(config, video_uri)
-            annotations_evaluated_features = evaluate_abcd_features_using_annotations(
-                config,
-                video_uri
-            )
-            video_assessment["annotations_evaluation"] = {
-                "evaluated_features": annotations_evaluated_features,
-            }
-
-        if config.use_llms:
-            llm_evaluated_features = evaluate_abcd_features_using_llms(
-                config, video_uri, prompt_params, llm_params
-            )
-            video_assessment["llms_evaluation"] = {
-                "evaluated_features": llm_evaluated_features,
-            }
-
-            if config.verbose:
-                if len(llm_evaluated_features) < len(get_feature_configs()):
-                    print(
-                        f"WARNING: ABCD Detector was not able to process all the features for video {video_uri}. Please check and execute again. \n"
-                    )
-                if len(llm_evaluated_features) > len(get_feature_configs()):
-                    print(
-                        f"WARNING: ABCD Detector processed more features than the original number features. \
-                    Processed features: {len(llm_evaluated_features)} - Original features: {len(get_feature_configs())}"
-                    )
-
-        print_abcd_assessment(config.brand_name, video_assessment)
-        brand_assessment.get("video_assessments").append(video_assessment)
-        if config.bq_table_name:
-            store_in_bq(config, bq_service, video_assessment, prompt_params, llm_params)
-        # Remove local version of video files
-        remove_local_video_files()
-
-    if config.assessment_file:
-        with open(config.assessment_file, "w", encoding="utf-8") as f:
-            json.dump(brand_assessment, f, ensure_ascii=False, indent=4)
-
-    return brand_assessment
+  Args:
+    arg_list: A list of command line arguments
+
+  """
+
+  args = parse_args(arg_list)
+
+  config = build_abcd_params_config(args)
```

Comment:
Change to config.from_args() and move both: parse_args(arg_list) and build_abcd_params_config(args) into that class.

Reply:
it is in-line with best practices as-is so would not recommend doing this.

Reply:
Responded in the other thread.